modules/fbs/includes/fbs.user.inc

```diff
@@ -62,6 +62,7 @@ function fbs_user_authenticate($name, $pincode) {
     // We have to save the library card number/CPR number, as the pincode
     // change call uses it.
     $result['creds']['library_card_number'] = $name;
+    $result['creds']['pass'] = $pincode;
```

Comment:
For starters I think it's bad form for anyone but the provider in use to dig around in $creds and making assumptions about it's structure.

This also lessens security for sites not using infomedia.

I'd rather se a ding_user_get_credentials() that infomedia can call which might throw up a password prompt if the password isn't available.